WordPress/src/main/java/org/wordpress/android/ui/stories/SaveInitialPostUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package org.wordpress.android.ui.stories
+
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.post.PostStatus
+import org.wordpress.android.fluxc.store.PostStore
+import org.wordpress.android.ui.posts.EditPostRepository
+import org.wordpress.android.ui.posts.SavePostToDbUseCase
+import javax.inject.Inject
+
+class SaveInitialPostUseCase @Inject constructor(
+    val postStore: PostStore,
+    val savePostToDbUseCase: SavePostToDbUseCase
+) {
+    fun saveInitialPost(editPostRepository: EditPostRepository, site: SiteModel?) {
+        editPostRepository.set {
+            val post: PostModel = postStore.instantiatePostModel(site, false, null, null)
+            post.setStatus(PostStatus.DRAFT.toString())
+            post
+        }
+        editPostRepository.savePostSnapshot()
+        // this is an artifact to be able to call savePostToDb()
+        editPostRepository.getEditablePost()?.setPostFormat(StoryComposerActivity.POST_FORMAT_WP_STORY_KEY)
```

Comment:
In general, it's preferred to work only with immutable posts and use `update/updateAsync` methods for modifications. Let me know if it'd work for you.

Reply:
Sure that works! I refactored this. Updated here 9bed509